bittensor/extrinsics/network.py

```diff
@@ -167,46 +174,44 @@ def set_hyperparameter_extrinsic(
-            )
-            value_argument = extrinsic_params["fields"][
-                len(extrinsic_params["fields"]) - 1
-            ]
-
-            # create extrinsic call
-            call = substrate.compose_call(
-                call_module="AdminUtils",
-                call_function=extrinsic,
-                call_params={"netuid": netuid, str(value_argument["name"]): value},
-            )
-            extrinsic = substrate.create_signed_extrinsic(
-                call=call, keypair=wallet.coldkey
-            )
-            response = substrate.submit_extrinsic(
-                extrinsic,
-                wait_for_inclusion=wait_for_inclusion,
-                wait_for_finalization=wait_for_finalization,
-            )
+        extrinsic_params = await subtensor.substrate.get_metadata_call_function(
+            "AdminUtils", extrinsic
+        )
+        value_argument = extrinsic_params["fields"][len(extrinsic_params["fields"]) - 1]
 
-            # We only wait here if we expect finalization.
-            if not wait_for_finalization and not wait_for_inclusion:
-                return True
-
-            # process if registration successful
-            response.process_events()
-            if not response.is_success:
-                bittensor.__console__.print(
-                    ":cross_mark: [red]Failed[/red]: error:{}".format(
-                        response.error_message
-                    )
-                )
-                time.sleep(0.5)
+        # create extrinsic call
+        call = await subtensor.substrate.compose_call(
+            call_module="AdminUtils",
+            call_function=extrinsic,
+            call_params={"netuid": netuid, str(value_argument["name"]): value},
+        )
+        extrinsic = await subtensor.substrate.create_signed_extrinsic(
+            call=call, keypair=wallet.coldkey
+        )
+        response = await subtensor.substrate.submit_extrinsic(
+            extrinsic,
+            wait_for_inclusion=wait_for_inclusion,
+            wait_for_finalization=wait_for_finalization,
+        )
 
-            # Successful registration, final check for membership
-            else:
-                bittensor.__console__.print(
-                    f":white_heavy_check_mark: [green]Hyper parameter {parameter} changed to {value}[/green]"
+        # We only wait here if we expect finalization.
+        if not wait_for_finalization and not wait_for_inclusion:
+            return True
+
+        # process if registration successful
+        # TODO: ? passably have to be awaited after `submit_extrinsic` implemented in `async_substrate.py`
+        response.process_events()
+        if not response.is_success:
+            bittensor.__console__.print(
+                ":cross_mark: [red]Failed[/red]: error:{}".format(
+                    response.error_message
                 )
-                return True
+            )
+            await asyncio.sleep(0.5)
```

Comment:
Why do we wait here? I know we do in the original also, but why?

Reply:
Honestly, I don't see any valuable reasons for this action. But I just convert this into async. Let's remove sleep at all?